backend/asmgen.ml

```diff
@@ -348,12 +348,28 @@ let compile_fundecl ~ppf_dump ~funcnames fd_cmm =
             Regalloc_validate.Description.create (Cfg_with_infos.cfg_with_layout cfg)
         in
         cfg
-        ++ begin match regalloc with
-          | GI -> Profile.record ~accumulate:true "cfg_gi" Regalloc_gi.run
-          | IRC -> Profile.record ~accumulate:true "cfg_irc" Regalloc_irc.run
-          | LS -> Profile.record ~accumulate:true "cfg_ls" Regalloc_ls.run
-          | Upstream -> assert false
-        end
+        ++
+        (
```

Comment:
This part of `Asmgen` is already hard to follow to get an overview of the pipeline. Is it possible to factor out counters into a separate function?